modules/thermal_hydraulics/include/materials/ADWallHeatTransferCoefficientWeismanMaterial.h

```diff
@@ -0,0 +1,58 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "Material.h"
+
+/**
+ * Computes wall heat transfer coefficient for liquid sodium using Schad-modified correlation
+ */
+class ADWallHeatTransferCoefficientWeismanMaterial : public Material
+{
+public:
+  ADWallHeatTransferCoefficientWeismanMaterial(const InputParameters & parameters);
+
+protected:
+  virtual void computeQpProperties();
+
+  /// Wall heat transfer coefficient
+  ADMaterialProperty<Real> & _Hw;
+  /// Density
+  const ADMaterialProperty<Real> & _rho;
+  /// Velocity
+  const ADMaterialProperty<Real> & _vel;
+  /// Hydraulic diameter
+  const ADMaterialProperty<Real> & _D_h;
+  /// Thermal conductivity
+  const ADMaterialProperty<Real> & _k;
+  /// Dynamic viscosity
+  const ADMaterialProperty<Real> & _mu;
+  /// Specific heat capacity
+  const ADMaterialProperty<Real> & _cp;
+  /// Fluid temperature
+  const ADMaterialProperty<Real> & _T;
+  /// Wall temperature
+  const ADMaterialProperty<Real> & _T_wall;
+  /// Pitch-to-Diameter ratio
+  const Real & _PoD;
+  /// Array type
+protected:
+  /// Rod bundle array type
+  enum class Bundle_array
```

Comment:
It's preferred to define your enums at the top of the access (`protected` in your case) section, for visibility.

Reply:
Ok, thanks, I will do it for now on!